starter_code/src/movies.js

```diff
@@ -1,20 +1,102 @@
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+// Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(moviesArray) {
+  return moviesArray.map(function(elem) {
+    var hours = 0;
+    var minutes = 0;
+    if (elem.duration.indexOf("h") !== -1) {
+      hours = parseInt(elem.duration[0], 10) * 60;
+    }
+    if (elem.duration.indexOf("min") !== -1) {
+      minutes = parseInt(
+        elem.duration.substring(
+          elem.duration.length - 5,
+          elem.duration.length - 3
+        ),
+        10
+      );
+    }
+    return Object.assign({}, elem, { duration: hours + minutes });
+  });
+}
+turnHoursToMinutes(movies);
+console.log(movies);
 
-// Get the average of all rates with 2 decimals 
+// Get the average of all rates with 2 decimals
 
+function ratesAverage(moviesArray) {
+  var totalRates = moviesArray.reduce(function(accumulator, current) {
+    return accumulator + current.rate;
+  }, 0);
+
+  var average = totalRates / moviesArray.length;
+
+  average = Math.round(average * 100) / 100;
+
+  return average;
+}
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate(moviesArray) {
+  var dramaMovies = moviesArray.filter(function(movie) {
+    return movie.genre.indexOf("Drama") > -1;
+  });
+
+  if (dramaMovies.length == 0) {
+    return undefined;
+  } else {
+    var result = ratesAverage(dramaMovies);
+    return Math.round(result * 100) / 100;
+  }
+}
 
 // Order by time duration, in growing order
```

Comment:
You need to keep in mind that the code runs and you already transformed all the durations in minutes and that what is asked here is to return "movieArray" in a growing order.
So first, you sort the durations keeping in mind the alphabetical order.
you return these durations then the movieArray.
This is a case of a function inside a function. 
Keep in mind the final objective but start from the smallest function.

function orderByDuration(moviesArray) {
    moviesArray.sort(function (a, b) {
      if (a.duration === b.duration) {
        if (a.title > b.title) {
          return 1;
        }
      }
      return a.duration - b.duration;
    });
    return moviesArray;
  }